genshin/client/manager.py

```diff
@@ -105,7 +118,7 @@ async def _request(
     ) -> typing.Any:
         """Make a request towards any json resource."""
         async with self.create_session() as session:
-            async with session.request(method, str_or_url, **kwargs) as response:
+            async with session.request(method, str_or_url, proxy=self.proxy, **kwargs) as response:
```

Comment:
If a proxy is going to be set per-request it also needs to be added for static requests in `BaseClient`